### PR TITLE
fix(plasma-hope): Handling correct offset 

### PR DESCRIPTION
**Pull request by Yakutoc**
### Проблема: 

При изменения позиции контейнера, где находиться компонент с `Tooltip` при его появление **рассчитывались неправильные отступы**.

#### Документация
 
[update-forceupdate-and-state](https://popper.js.org/react-popper/v2/hook/#update-forceupdate-and-state)

[Instance](https://popper.js.org/docs/v2/constructors/#instance)   

### Было:
Screen Recording 2023-04-03 at 19 31 41

### Стало:

Screen Recording 2023-04-03 at 19 37 05

:baby_chick: Download canary assets:

[colors--canary.461.4604191287.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.461.4604191287.xml)  
[color_sberHealth_ios-swift--canary.461.4604191287.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.461.4604191287.swift)  
[color_sberHealth_kotlin--canary.461.4604191287.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.461.4604191287.kt)  
[color_sberHealth_react-native--canary.461.4604191287.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.461.4604191287.ts)  
[color_sbermarket_business_ios-swift--canary.461.4604191287.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_ios-swift--canary.461.4604191287.swift)  
[color_sbermarket_business_kotlin--canary.461.4604191287.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_kotlin--canary.461.4604191287.kt)  
[color_sbermarket_business_react-native--canary.461.4604191287.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_react-native--canary.461.4604191287.ts)  
[color_sbermarket_ios-swift--canary.461.4604191287.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.461.4604191287.swift)  
[color_sbermarket_kotlin--canary.461.4604191287.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.461.4604191287.kt)  
[color_sbermarket_metro_ios-swift--canary.461.4604191287.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_ios-swift--canary.461.4604191287.swift)  
[color_sbermarket_metro_kotlin--canary.461.4604191287.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_kotlin--canary.461.4604191287.kt)  
[color_sbermarket_metro_react-native--canary.461.4604191287.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_react-native--canary.461.4604191287.ts)  
[color_sbermarket_react-native--canary.461.4604191287.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.461.4604191287.ts)  
[color_sbermarket_selgros_ios-swift--canary.461.4604191287.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_ios-swift--canary.461.4604191287.swift)  
[color_sbermarket_selgros_kotlin--canary.461.4604191287.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_kotlin--canary.461.4604191287.kt)  
[color_sbermarket_selgros_react-native--canary.461.4604191287.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_react-native--canary.461.4604191287.ts)  
[color_sbermarket_vinlab_ios-swift--canary.461.4604191287.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_vinlab_ios-swift--canary.461.4604191287.swift)  
[color_sbermarket_vinlab_kotlin--canary.461.4604191287.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_vinlab_kotlin--canary.461.4604191287.kt)  
[color_sbermarket_vinlab_react-native--canary.461.4604191287.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_vinlab_react-native--canary.461.4604191287.ts)  
[color_sberprime_ios-swift--canary.461.4604191287.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.461.4604191287.swift)  
[color_sberprime_kotlin--canary.461.4604191287.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.461.4604191287.kt)  
[color_sberprime_react-native--canary.461.4604191287.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.461.4604191287.ts)  
[PlasmaTokensColor--canary.461.4604191287.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.461.4604191287.swift)  
[shadow_sbermarket_react-native--canary.461.4604191287.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.461.4604191287.ts)  
[typo_mage_ios-swift--canary.461.4604191287.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.461.4604191287.swift)  
[typo_mage_kotlin--canary.461.4604191287.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.461.4604191287.kt)  
[typo_mage_react-native--canary.461.4604191287.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.461.4604191287.ts)  
[typo_plasma_ios-swift--canary.461.4604191287.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.461.4604191287.swift)  
[typo_plasma_kotlin--canary.461.4604191287.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.461.4604191287.kt)  
[typo_plasma_react-native--canary.461.4604191287.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.461.4604191287.ts)  
[typo_ruler_ios-swift--canary.461.4604191287.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.461.4604191287.swift)  
[typo_ruler_kotlin--canary.461.4604191287.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.461.4604191287.kt)  
[typo_ruler_react-native--canary.461.4604191287.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.461.4604191287.ts)  
[typo_sage_ios-swift--canary.461.4604191287.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.461.4604191287.swift)  
[typo_sage_kotlin--canary.461.4604191287.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.461.4604191287.kt)  
[typo_sage_react-native--canary.461.4604191287.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.461.4604191287.ts)  
[typo_sbermarket_ios-swift--canary.461.4604191287.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.461.4604191287.swift)  
[typo_sbermarket_kotlin--canary.461.4604191287.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.461.4604191287.kt)  
[typo_sbermarket_react-native--canary.461.4604191287.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.461.4604191287.ts)  
[typo_soulmate_ios-swift--canary.461.4604191287.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.461.4604191287.swift)  
[typo_soulmate_kotlin--canary.461.4604191287.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.461.4604191287.kt)  
[typo_soulmate_react-native--canary.461.4604191287.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.461.4604191287.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.173.1-canary.461.4604191287.0
  npm install @salutejs/plasma-hope@0.31.1-canary.461.4604191287.0
  npm install @salutejs/plasma-temple@1.148.1-canary.461.4604191287.0
  npm install @salutejs/plasma-ui@1.180.1-canary.461.4604191287.0
  npm install @salutejs/plasma-web@1.198.1-canary.461.4604191287.0
  # or 
  yarn add @salutejs/plasma-b2c@1.173.1-canary.461.4604191287.0
  yarn add @salutejs/plasma-hope@0.31.1-canary.461.4604191287.0
  yarn add @salutejs/plasma-temple@1.148.1-canary.461.4604191287.0
  yarn add @salutejs/plasma-ui@1.180.1-canary.461.4604191287.0
  yarn add @salutejs/plasma-web@1.198.1-canary.461.4604191287.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-461/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-461/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-461/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-461/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-461/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-461/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-461/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-461/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Review by neretin-trike — APPROVED** [review]

File: packages/plasma-hope/src/components/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -147,6 +149,14 @@ export const Tooltip = forwardRef<HTMLSpanElement, TooltipProps>(
             };
         }, []);
 
+        useEffect(() => {
+            if (!isVisible || !forceUpdate) {
```
Comment (Yeti-or) [inline packages/plasma-hope/src/components/Tooltip/Tooltip.tsx:153]: forceUpdate может быть null ??

Comment (Yakutoc) [reply packages/plasma-hope/src/components/Tooltip/Tooltip.tsx:153]: @Yeti-or yep ) 

> These properties match the ones described in the [Popper docs](https://popper.js.org/docs/v2/constructors/#types), the only difference is that they **can be null** if Popper isn't yet been initialized or has been destroyed.

**Review by Yeti-or — APPROVED** [review]

**Pull request merged**